index.js

```diff
@@ -154,18 +151,33 @@ export class SDK extends EventEmitter {
       const discovery = this.join(core.discoveryKey, opts)
       core.discovery = discovery
 
+      // If we're the owner, then we could wait until is fully announced
+      if (core.writable) {
+        // await discovery.flushed()
```

Comment:
If you uncomment this, then you don't need the retry logic below. If that makes sense to you then let me know and I remove the retry part!

Reply:
Both things could be enabled as well, flushed() for the server, and retry for the client

Reply:
That's perfect, thank you for the PR! Gonna test it out ASAP.